changelogs/fragments/839-Add-Field-to-zos-job-query.yml

```diff
@@ -0,0 +1,4 @@
+minor_changes:
```

Comment:
Its tedious but changelog won't like this, it should really be separate entries into the one changlog fragment:
```
minor_changes:
- zos_job_query - zoau added 'program_name' to their field output starting with v1.2.4. 
  This enhancement checks for that version and passes the extra column through.
  (https://github.com/ansible-collections/ibm_zos_core/pull/841)
- zos_job_submit - zoau added 'program_name' to their field output starting with v1.2.4. 
   This enhancement checks for that version and passes the extra column through.
  (https://github.com/ansible-collections/ibm_zos_core/pull/841)
- zos_job_output - zoau added 'program_name' to their field output starting with v1.2.4. 
  This enhancement checks for that version and passes the extra column through.
  (https://github.com/ansible-collections/ibm_zos_core/pull/841)
```

Reply:
live and learn... I'm still learning